experimental/casemapping/src/provider.rs

```diff
@@ -0,0 +1,33 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use crate::internals::CaseMappingInternals;
+use icu_provider::yoke;
+
+pub mod key {
+    //! Resource keys for [`icu_casemapping`](crate)
+    use icu_provider::{resource_key, ResourceKey};
+
+    /// Key for case mapping data
+    pub const CASE_MAPPING_V1: ResourceKey = resource_key!("props/casemap@1");
+}
```

Comment:
Nit: You can and should remove the `key` module now that the key is in the ResourceMarker.